db/query.ts

```diff
@@ -1,54 +1,227 @@
-import pg, { QueryResult } from 'pg';
+import { Pool, QueryResult } from 'pg';
 import fs from 'node:fs';
 import path from 'node:path';
 import jsonToCsv from './export';
+import { ALL_OUTGOING, ALL_INCOMING } from './queries/all';
 import Row from './row';
-
+import { AccountId, FTBalance, getCurrencyByContractFromNear, getFTBalance } from '../helpers/currency';
+import { getLockup } from '../helpers/lockup';
 const CONNECTION_STRING = process.env.POSTGRESQL_CONNECTION_STRING;
 
-// TODO: Consider allowing these values to be configurable per environment:
-const STATEMENT_TIMEOUT = 30 * 1_000; // 30 seconds in milliseconds. "number of milliseconds before a statement in query will time out" https://node-postgres.com/api/client
+const SQL_STATEMENT_TIMEOUT = 3600 * 1_000; // 1 hour in milliseconds.
+
+export default async function query_all(startDate: string, endDate: string, accountIds: Set<string>) {
+  const pool = new Pool({ connectionString: CONNECTION_STRING, statement_timeout: SQL_STATEMENT_TIMEOUT });
+  let rows_promises = [];
+
+  console.log('query_all', startDate, endDate, accountIds);
+
+  for (const accountId of accountIds) {
+    const lockupAccountId = getLockup('near', accountId);
+
+    const all_outgoing_txs_promise = pool.query(ALL_OUTGOING, [[accountId, lockupAccountId], startDate, endDate]);
+    const all_incoming_txs_promise = pool.query(ALL_INCOMING, [[accountId, lockupAccountId], startDate, endDate]);
+    const [all_outgoing_txs, all_incoming_txs] = await Promise.all([all_outgoing_txs_promise, all_incoming_txs_promise]);
 
-const sqlFolder = path.join(path.join(process.cwd(), 'db'), 'queries');
-const DOT_SQL = '.sql';
+    // TODO(pierre): consider using async to parallelize this
+    for (const row of all_outgoing_txs.rows) {
+      rows_promises.push(handleOutgoing(accountId, row));
+    }
 
-function getTransactionTypeSql(file: string): string {
-  const sql = fs.readFileSync(file, 'utf8');
-  return sql;
+    for (const row of all_incoming_txs.rows) {
+      rows_promises.push(handleIncoming(accountId, row));
+    }
+  }
+
+  let rows = await Promise.all(rows_promises);
+
+  const sortedRows = sortByBlockTimestamp(rows);
+  const csv = jsonToCsv(sortedRows);
+  return csv;
 }
 
-function getFiles() {
-  const filesUnfiltered = fs.readdirSync(sqlFolder);
-  console.log({ filesUnfiltered });
-  return filesUnfiltered.filter((file) => file.endsWith(DOT_SQL)).map((file) => path.join(sqlFolder, file));
+function formatDate(date: Date): string {
+  const year = date.getFullYear();
+  const month = date.getMonth() + 1;
+  const day = date.getDate();
+  return `${day}/${month}/${year}`;
 }
 
-function sortByBlockTimestamp(rows: Row[]): Row[] {
-  return rows.sort((a, b) => a.block_timestamp - b.block_timestamp);
+async function handleOutgoing(accountId: AccountId, row: any): Promise<Row> {
```

Comment:
This function is longer than we want, so our eslint rules will flag it. 

It also has a bunch of opportunity to reduce duplication (e.g. `(raw_amount_out / 10 ** decimals)`) so I'll try to make some adjustments in my upcoming commit.

We'll also generally want to have more descriptive, self-documenting names than `handleX()`.

Reply:
Agree here but I think we should focus on correct functionality first, then we can optimize ! Both for code patterns and naming